src/main/java/com/patchworkmc/mapping/remapper/PatchworkRemapper.java

```diff
@@ -0,0 +1,193 @@
+package com.patchworkmc.mapping.remapper;
+
+import java.util.HashMap;
+import java.util.HashSet;
+
+import net.fabricmc.tinyremapper.IMappingProvider;
+
+import net.patchworkmc.manifest.accesstransformer.v2.exception.MissingMappingException;
+
+import com.patchworkmc.Patchwork;
+
+public class PatchworkRemapper {
```

Comment:
What's the rationale for not using tiny-remapper's Remapper here? (For PatchworkRemapper itself, not PatchworkRemapper.Naive)

Reply:
A couple reasons:
- TR's AsmRemapper returns `null` instead of having checked exceptions
-  A TR instance is more expensive to build and requires closing
- We already have to load mappings here for NaiveRemapper, so this saves even more time by reducing the amount of times we load the mappings.

Reply:
My main issue is that we duplicate a bunch of logic probably already contained in tiny-remapper with this new PatchworkRemapper class. I'd much prefer to use the existing code within tiny-remapper instead since it reduces the amount of potential bugs significantly: tiny-remapper is heavily tested. If at all possible I would prefer to use tiny-remapper for these reasons.

Reply:
Oh, and this avoids a NaiveRemapper call because ignorefielddesc just... wasn't working or didn't do what the name implies it does

Reply:
Will open an issue when complete